sway/desktop/render.c

```diff
@@ -1998,9 +1892,17 @@ void output_render(struct sway_output *output, struct timespec *when,
 		render_layer_toplevel(output, damage,
 			&output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BOTTOM]);
 
-		bool blur_enabled = has_blur && should_parameters_blur();
-		if (blur_enabled && blur_optimize_should_render && renderer->blur_buffer_dirty) {
-			render_monitor_blur(output, damage);
+		// check if the background needs to be blurred
+		if (should_parameters_blur() && renderer->blur_buffer_dirty) {
+			should_render_blur = should_workspace_have_blur(workspace);
```

Comment:
TODO: right now, this just checks if a workspace has blur, not if that blur should be optimized